### ISSUE TITLE
Fields that seems to suggest a finite list of choices

### ISSUE THREAD

**Issue by evz** [security-force-monitor/sfm-cms#26]
Some fields suggest that there is a finite list of choices that should be available for a user to be able to enter. For these fields, we could include initial sources of data that we could load in when setting up the app and it would cut down a bit on the complexity of things under the hood. These fields include:

- [ ] Rank & Role for the MembershipPerson data type
- [ ] Classifications for the Organization data type
- [ ] Violation type for the Violation data type

There may be others. @tonysecurityforcemonitor Are these something that we can pre-poulate? Or will we need to be dynamically creating these based upon user input?

**Comment by tonysecurityforcemonitor** [comment]
@evz we can pre-populate - let me break up the different fields though because they are actually all a little different

- [ ] Rank - in theory once we know the rank structure of a country it rarely (if ever) changes - so pre-populating would be really helpful
- [ ] Role - the only role that really has "meaning" is Commander - everything else should be dynamic to allow the researcher to copy/past the role as is from the source, rather than making decisions about it (ie is a "spokesperson" a Public Relations Officer which is an official title/role? maybe, maybe not --- either way not good use of time to try to tease that out)
- [ ] Classifications for the Organization data type - this should be the least "dynamic" - as the front-end UI should be using these classifications for the filter for the map and other screens so we want to keep the list as short and meaningful as possible. It would actually be good if there was only a pre-populated list that only certain users (like me) could modify
- [ ] Violation type for the Violation data type --- as far as I know this should be pre-populated and I don't think the list is being modified anytime soon @tlongers - this is based on HURIDOCS standards so I think the top level types of acts aren't changing anytime soon (right?)

**Comment by tlongers** [comment]
re: "this is based on HURIDOCS standards so I think the top level types of acts aren't changing anytime soon (right?)" - unlikely in near term. I don't think an overhaul is underway.

**Comment by tonysecurityforcemonitor** [comment]
@evz - I forgot to add - Role, Rank, Classifications for the Organization data type, and Violation type for the Violation data type should all have matching translations. I believe I shared the Google Sheet with the current running list.

I think that means we might be better served by having a more controlled vocabulary for each field that forces myelf/other staff and the inputter to confer before they add something - or otherwise make sure we are signing off in someway so that there aren't random entries with no translations floating around...

For Violation type for the Violation data type that list is fixed as are the translations.

**Comment by evz** [comment]
@tonysecurityforcemonitor For Rank, would it make sense to also store a country? It seems like it might be simplest if we just make this into an autocomplete similar to the one we have for organization name that looks for matching values and if it can't find anything, it'll create the new value. Does that make sense?

**Comment by tonysecurityforcemonitor** [comment]
@evz I don't think we need to attach the country to the rank --- ranks are actually very standard across all countries - there are a few variations but there is a "core" group of ranks that every country uses (Colonels for instance).

**Comment by tlongers** [comment]
re: match or create new option for Rank -  @evz Is there currently something there to let admin add/edit control  controlled vocabs or user-created category lists?

**Comment by evz** [comment]
@tlongers As of right now there isn't a simple way of doing that. I've just been creating files that we can easily import into the database as part of the setup (this is something that the web framework we're using has built into it). So, for example, the one that I made for violation types [is here](https://github.com/security-force-monitor/sfm-cms/blob/master/violation/fixtures/violation_types.json). 

As an alternative to this, there is an admin interface that comes out of the box as part of the web framework. This would allow an admin to have a rudimentary interface to add and edit things like ranks and roles. Perhaps the next time we have a checkin call, I can walk you through what that looks like and maybe that will work for the moment.
